### PR TITLE
Issue 2378 - Show TLS information for routes, create routes and routes/routename pages

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://github.com/openshift/origin/issues/2378

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett @liggitt 

I didn't want to dump all the cert / key info directly on the page, so i hid them behind click-to-reveal, maybe there is better link text than "Show"

**Comment by jwforres** [comment]
For routes that don't have TLS settings at all, hide the whole section?  Leave the header with a message that says TLS is not enabled for this route?

**Comment by jwforres** [comment]
And actually changing the pre tag to use inline-block instead of block looks better i think:

**Comment by jwforres** [comment]
fyi dont merge yet, adding path and port still

**Comment by jwforres** [comment]
ok path and port are included now

File: assets/app/views/browse/routes.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                <span ng-if="!(route | isWebRoute)">
+                  {{route | routeLabel}}
+                </span>
+              </td>
+              <td>
+                <span>{{route.spec.to.kind}}: </span>
+                <span ng-if="route.spec.to.kind != 'Service'">{{route.spec.to.name}}</span>
+                <span ng-if="route.spec.to.kind == 'Service'"><a href="project/{{projectName}}/browse/services/{{route.spec.to.name}}">{{route.spec.to.name}}</a></span>
+              </td>
+              <td><relative-timestamp timestamp="route.metadata.creationTimestamp" drop-suffix="true"></relative-timestamp></td>
```
Comment (spadgett) [inline assets/app/views/browse/routes.html]: I'm not sure how useful creation time is in these tables. I realize we have it in other places.

**Comment by spadgett** [comment]
> For routes that don't have TLS settings at all, hide the whole section? Leave the header with a message that says TLS is not enabled for this route?

I like having a message saying TLS is not enabled.

File: assets/app/views/browse/route.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<div ng-controller="RouteController" class="content">
+  <project-page>
+    <breadcrumbs breadcrumbs="breadcrumbs"></breadcrumbs>
+    <alerts alerts="alerts"></alerts>
+    <div ng-if="route">
+      <div class="col-md-12">
```
Comment (spadgett) [inline assets/app/views/browse/route.html]: Shouldn't `col-md-*` be inside a `row`?

File: assets/app/views/browse/route.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        <div class="tile">
+          <dl class="dl-horizontal left">
+            <dt>Name:</dt>
+            <dd>{{route.metadata.name}}</dd>
+            <dt>Labels:</dt>
+            <dd>
+              <span ng-if="!route.metadata.labels">none</span>
```
Comment (spadgett) [inline assets/app/views/browse/route.html]: Is `route.metadata.labels` an object? `(route.metadata.labels | hashSize) === 0`?

File: assets/app/views/browse/route.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<div ng-controller="RouteController" class="content">
+  <project-page>
+    <breadcrumbs breadcrumbs="breadcrumbs"></breadcrumbs>
+    <alerts alerts="alerts"></alerts>
+    <div ng-if="route">
```
Comment (spadgett) [inline assets/app/views/browse/route.html]: Do we need a loading message?

File: assets/app/views/browse/route.html [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      <div class="col-md-12">
+        <div class="tile">
+          <dl class="dl-horizontal left">
+            <dt>Name:</dt>
+            <dd>{{route.metadata.name}}</dd>
+            <dt>Labels:</dt>
```
Comment (spadgett) [inline assets/app/views/browse/route.html]: Missed annotations?

File: assets/app/views/browse/route.html (hunk 2 above)
Comment (jwforres) [inline assets/app/views/browse/route.html]: Yes. Copy/paste fail.

File: assets/app/views/browse/route.html (hunk 3 above)
Comment (jwforres) [inline assets/app/views/browse/route.html]: It is but it is set to null when empty so the key isn't even in the api
response

File: assets/app/views/browse/route.html (hunk 4 above)
Comment (jwforres) [inline assets/app/views/browse/route.html]: Looks like I missed that on all of the individual pages, I'm adding a todo on the trello card to add it for all of them. (will do it in a separate PR)

File: assets/app/views/browse/route.html (hunk 5 above)
Comment (jwforres) [inline assets/app/views/browse/route.html]: Same thing here, adding a specific todo for this.  Want to make sure annotations are treated consistently across the pages.

**Comment by jwforres** [comment]
[merge]

**Comment by jwforres** [comment]
[merge]
